app/tools/research_tools.py

```
@@ -167,19 +194,77 @@ def search(self, query: str, max_results: int = 10, include_images: bool = True)
+                        result = self.client.search(
+                            query=f"site:{url}",
+                            max_results=1,
+                            include_answer=True,
+                        )
+                        if result.get("results"):
+                            all_results.append({
+                                "url": url,
+                                "content": result["results"][0].get("content", ""),
+                                "raw_content": result["results"][0].get("raw_content", ""),
+                            })
+                    except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)